internal/config/analytics.go

```diff
@@ -0,0 +1,61 @@
+package config
+
+import (
+	"errors"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/spf13/viper"
+)
+
+// AnalyticsConfig defines the configuration for various mechanisms for
+// reporting and querying analytical data for Flipt.
+type AnalyticsConfig struct {
+	Enabled    bool             `json:"enabled,omitempty" mapstructure:"enabled" yaml:"enabled,omitempty"`
+	Clickhouse ClickhouseConfig `json:"clickhouse,omitempty" mapstructure:"clickhouse" yaml:"clickhouse,omitempty"`
+	Buffer     BufferConfig     `json:"buffer,omitempty" mapstructure:"buffer" yaml:"buffer,omitempty"`
+}
+
+// ClickhouseConfig defines the connection details for connecting Flipt to Clickhouse.
+type ClickhouseConfig struct {
+	Enabled  bool   `json:"enabled,omitempty" mapstructure:"enabled" yaml:"enabled,omitempty"`
+	URL      string `json:"url,omitempty" mapstructure:"url" yaml:"url,omitempty"`
+	Username string `json:"username,omitempty" mapstructure:"username" yaml:"username,omitempty"`
+	Password string `json:"password,omitempty" mapstructure:"password" yaml:"password,omitempty"`
+}
+
+func (c *ClickhouseConfig) Auth() clickhouse.Auth {
+	return clickhouse.Auth{
+		Username: c.Username,
+		Password: c.Password,
+		Database: "flipt_analytics",
```

Comment:
is the db part of the url? or do we 'own' the db name?

Reply:
@markphelps Yeah the idea is that we 'own' the db name, basically that the users would have to create the DB beforehand.

I believe the DB could be part of the URL but the API that we are using separates these two concepts: https://pkg.go.dev/github.com/beebeeep/clickhouse-go/v2#Options

Reply:
So it looks like the provide a [ParseDSN](https://pkg.go.dev/github.com/beebeeep/clickhouse-go/v2#ParseDSN) function that will take care of providing the clickhouse options for you. I might go that route instead.

But within the `Options` itself, they separate the concept of `Host` and `Username`/`Password`.

If you provide a username and password for an `Addr`, it actually errors out.